source/extensions/config_subscription/grpc/grpc_mux_failover.h

```diff
@@ -221,11 +222,11 @@ class GrpcMuxFailover : public GrpcStreamInterface<RequestType, ResponseType>,
                              "in a row. Attempting to connect to the failover stream.");
             // This will close the stream and prevent the retry timer from
             // reconnecting to the primary source.
-            // TODO(adisuissa): need to ensure that when moving between primary and failover,
-            // the initial_resource_versions that are sent are empty. This will be
-            // done in a followup PR.
             parent_.primary_grpc_stream_->closeStream();
-            parent_.grpc_mux_callbacks_.onEstablishmentFailure();
+            // Next attempt will be to the failover, set the value that
+            // determines whether to set initial_resource_versions or not.
+            parent_.grpc_mux_callbacks_.onEstablishmentFailure(parent_.previously_connected_to_ ==
```

Comment:
I thought we always sent initial resource version when connecting to the failover?

Reply:
At the moment the failover is only being attempted if no resources were fetched from the primary (thus no initial versions should exist).

Reply:
Maybe a good candidate for an ASSERT?

Reply:
To be clear the code line mentioned above handles a case where
1. Envoy connected to failover (and fetches some resources)
2. Envoy is disconnected from the failover and attempts to connect to the primary - no initial_resource_versions will be sent.
3. Envoy never connects to the primary and attempts to connect to the failover - will send the initial_resource_versions that were fetched in (1).

RE the assert - there's an assertion that no failover stream is established when `ever_connected_to_primary_` is true.